### PR TITLE
Fix #54 by checking the remote on wa update

### LINKED ISSUE #54: 'wa update' does not 'git set-url upstream' to the updated repo urls
which leaves me with `upstream` set to
```
upstream	git@github.com:bucaran/wahoo.git (fetch)
upstream	git@github.com:bucaran/wahoo.git (push)
```

### PR DESCRIPTION
**Pull request by s0r00t**
The commented part might be removed; If the contributor had updated code before updating, there might be a problem.

~~I'll look onto this and commit if necessary.~~
Latest commit fixed this.

### CODE REVIEW

File: pkg/wa/cli/wa_update.fish [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
-  test -z (git config --get remote.upstream.url)
-    and set -l repo "origin"
-
+  set -l mainrepo "https://github.com/fish-shell/wahoo.git"
+  if [ (git config --get remote.origin.url) = $mainrepo ]
+    set repo "origin"
+  else
+    if not git remote|grep "upstream" -q
+      git remote add upstream $mainrepo
+      set repo "upstream"
```
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: local scope would be fine, `set -l repo`, same for the following `set`s

File: pkg/wa/cli/wa_update.fish [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+      git fetch upstream
+      git checkout master
+      if not git diff --exit-code > /dev/null
+        git merge upstream/master
+      else
+        git rebase upstream/master
+      end
+      git push -f origin master
```
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: `force` should not be used in general, and i feel like especially not in this case

File: pkg/wa/cli/wa_update.fish [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+    else
+      if [ (git config --get remote.upstream.url) = $mainrepo ]
```
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: please use portable `test` syntax like `-eq` and so

**Comment by s0r00t** [comment]
Local scope was the cause of the problem: if you test, you'll see that the variable doesn't exists out of the conditional block.

File: pkg/wa/cli/wa_update.fish [hunk 4]
```diff
@@ -1,8 +1,29 @@
...
+        set repo "upstream"
+      else
+        git remote set-url upstream $mainrepo
+        set repo "upstream"
```
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: three times same call of `set repo "upstream"` - recomming some simpler logic

File: pkg/wa/cli/wa_update.fish [hunk 5]
```diff
@@ -1,8 +1,29 @@
...
+      end
+    end
+  end
+      
   if git diff-index --quiet HEAD -- >/dev/null ^&1
```
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: i get theck check, but it is waht `git pull` does anyway, so it's a bit duped

Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: i get theck check, but it is waht git pull does anyway, so it's a bit duped

File: pkg/wa/cli/wa_update.fish (hunk 4 above)
Comment (s0r00t) [inline pkg/wa/cli/wa_update.fish]: Alright, taking care of that. If you have some idea of code give it on Gitter.

File: pkg/wa/cli/wa_update.fish (hunk 5 above)
Comment (s0r00t) [inline pkg/wa/cli/wa_update.fish]: What? I do not understand.

**Comment by gretel** [comment]
thanks for the PR, i appreciate it!
but personally i can currently only refuse the PR because i really don't know how this code should work if somebody decides to have remotes named differently. it should be up to the developer to set the remotes - and i usually do that lots. in general, there are no conventions on how to have remotes so i don't think `wahoo` should imply that.

**Comment by s0r00t** [comment]
Well, upstream remote is some kind of a convention. I rarely (if never) see developers not using upstream remote for their forks.

**Comment by gretel** [comment]
please point me to this supposed convention then

File: pkg/wa/cli/wa_update.fish (hunk 4 above)
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: i fell this place is just right because gitter is for chatting  :)

**Comment by s0r00t** [comment]
https://help.github.com/articles/configuring-a-remote-for-a-fork/
It's not a proper convention but it's something adopted by a lot of developers.

File: pkg/wa/cli/wa_update.fish (hunk 5 above)
Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: `git pull` implies `git diff-index` because otherwise git wouldn't 'know' about that there is something to pull remotely

**Comment by gretel** [comment]
so could we agree on the fact that it is a recommendation but that git does _per_ _design_ not imply any convention on how to name remotes? you could also ask why it is possible that names can be chosen freely. `upstram` is just another remote, important is what it refers to.

**Comment by s0r00t** [comment]
Sure. If you really want to support alternative names for the original repo remote then you could add support over (for example) a shell variable containing the name of the remote.

**Comment by gretel** [comment]
but why should i want to do that to compensate fror changes i dont get at all :)

**Comment by s0r00t** [comment]
"if i would like that 'stick with recommendations' pattern i'd possible use bash :dart:"

FYI you reported the issue from which this PR comes from (#54), and you are using upstream remotes.
So you're 'sticking with recommandations'(sic).

**Comment by gretel** [comment]
well instead of fixing a bug you are proposing changes that i don't follow.
again, how would you know what remotes i do use? got a backdoor on my machine i didn't notice? i also recommend doing testing first, because it think it's strange to mention "can't test" in a PR.

**Comment by s0r00t** [comment]
The tests were done.
How do I know which remotes you use? Well, easy : you said it in the issue. [Proof?](http://i.imgur.com/XgtctjL.png)

**Comment by gretel** [comment]
i guess you do realize that you should know a bit more about how git works before trying to proof that this issue seem to be so important for you that you actually have screenshot done and uploaded for stuff that i publicly visible on github.
won't get into any further discussion with you because i feel it's controversial at least.
if you feel offended by my comments i am sorry, honestly, but that won't make me think any different about your code being the actual subject of this PR.
regards and good fishing, tom!

**Comment by gretel** [comment]
@bucaran technically, going down the `git` road, i do suggest specs for the project structure in regards of the _dependencies_ we impose on environments like `github`. i would be fine having 'hardcoded' remote names like for the kinda-conventionalized `upstream` to support forking and eventually getting pull requests - but we should think about how we initially encourage people to have (rebased) feature-branches so we can handle PR smoothly. at the same time, we should try to not confuse people.
so summing up the facts i personally come to the following conclusion:
* specs in general have the highest priority for the project as of now
* second highest has extending and improving the documentation
* personally, i feel like 'code less, spec more' :haircut:

File: pkg/wa/cli/wa_update.fish (hunk 5 above)
Comment (s0r00t) [inline pkg/wa/cli/wa_update.fish]: Right, but I did not write the part using git diff-index.

Comment (gretel) [inline pkg/wa/cli/wa_update.fish]: if any i was blaming the code

**Comment by s0r00t** [comment]
I know enough about how git works.
The issue itself is not that much important : however, this is my work and I'd rather defend myself when you claim I took over your computer to prove your argument wrong. I took a screenshot to defend my claims.
You denied the PR for a reason you can't defend. It's not even linked to the code, it's linked to whether people are using a different remote to update the fork.
You contradicted yourself while defending why should this PR be denied. 
I proposed a solution for your concerns about the PR. Did you see it? I think not.

I don't have time for drama. Closing the PR.

**Comment by gretel** [comment]
i did not deny anything. in the contrary, i do let @bucaran decide on PRs - have a look on some other ones if you doubt that.
talking about definitions i have no clue of what you mean by "know enough".
when you do `git remote add` or `set-url` are you required to commit and/or push any changes then?
contradicted is your refusal of technical facts. the remotes are not stored remotely, which is the profound nature of this concept. without even thinking about the actual implementation that should make some sense to you.
it's the first time that i did report an issue on github and somebody obviously not even willing to understand what the actual issue is about trying to push code unconditionally. and i don't think it is going to happen anytime soon again.
after all, i hope we will still hang fine on gitter. regards

**Comment by s0r00t** [comment]
"when you do git remote add or set-url are you required to commit and/or push any changes then?"
No you don't. But after a rebase you might need to push the changes to the repo over GitHub.
"the remotes are not stored remotely"
I do know that remotes are not stored remotely. Isn't that a bit obvious?
"contradicted is your refusal of technical facts"
 When I say contradicted, you said that you weren't following recommandations such as using upstream as the name of the original remote. However, you do follow recommandations, because you use a remote named upstream.
This is all about using a goddamn word as a remote name.
I do not understand part of the end of your answer.
My work is not worth a fight. False clamings that I hijacked your computer does.

As I'm writing I'm working onto rewriting the code based on your comments. You still did not gave your word about the idea to have a variable containing the remote name.

**Comment by gretel** [comment]
> got a backdoor on my machine i didn't notice?
> False clamings that I hijacked your computer does.
:hankey:

**Comment by s0r00t** [comment]
:sweat: I did not.
Anyways, reopening.
So, list of incoming changes :
-Replace conditions by `test` when needed
-Remove push 

Is that all?

**Comment by bucaran** [comment]
We are getting lost between words. I am not really following this. Let's debate with algorithms. 

@gretel Please → #59

**Comment by s0r00t** [comment]
I just noticed I forgot to make a feature branch.
Sorry, I'll fix it if needed.

**Comment by gretel** [comment]
what is up with #59 please

**Comment by gretel** [comment]
as in https://github.com/fish-shell/wahoo/pull/62 i do suggest having no forking done
* less complexity
* less ambigious codepaths
* less confusion (user's choice)

**Comment by bucaran** [comment]
@s0r00t I am sorry, this thread became to complex to follow so, there are two options:

1. Please explain exactly this PR does in a reply :smile: Regarding the code, please add some one-liner comments, rebase, squash and push again.

2. Close this PR, create a new one.
